packages/calcite-components/src/components/tile-group/tile-group.tsx

```diff
@@ -79,13 +191,58 @@ export class TileGroup implements InteractiveComponent {
 
   disconnectedCallback(): void {
     disconnectInteractive(this);
+    this.mutationObserver?.disconnect();
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Event Listeners
+  //
+  //--------------------------------------------------------------------------
+
+  @Listen("calciteInternalTileKeyEvent")
+  calciteInternalTileKeyEventListener(event: CustomEvent): void {
+    if (event.composedPath().includes(this.el)) {
```

Comment:
[Per our guidelines](https://github.com/Esri/calcite-design-system/blob/main/packages/calcite-components/conventions/README.md#privateinternal-events), custom events should be stopped after handling.